### PR TITLE
#6523: Sync timeline guide layers

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds commits to improve the timeline guide layer sync.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
#6523 

**What is the new behavior?**
- Added new config option to Timeline
  -  If `showHiddenLayers: true`, the functionality stays as is
  -  If `showHiddenLayers: false` **(default)**, listens to two layer events _'remove/ hide layer'_, if time layer is removed/ hidden, the guide layer will not be displayed in the timeline and furthermore the timeline will switch to next available guide layer when **Snap to guide layer** option is enabled in Timeline settings
- **Snap to guide layer** setting will be preserved between the layer event changes

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by offtherailz — COMMENTED** [review]
Can you document the flag also in the JSDoc please?

File: docs/user-guide/timeline.md [hunk 1]
```diff
@@ -56,6 +56,13 @@ Sometimes you might be interested to show in the timeline histogram only the tim
 !!! Note
     Map Sync feature need at least GeoServer 2.15.2
 
+## Show hidden layers
+
+When then timeline configuration`showHiddenLayer`is set to *false* *(false by default)*, the layers in timeline gets in sync with time layer's visibility (TOC) i.e when a time layer is hidden or removed, the timeline will not show the respective guide layer. Furthermore, the timeline automatically selects the next available guide layer
+if the **Snap to guide layer** option is enabled in the Timeline settings. If set to *true*, the hidden layer will be shown in the timeline.
+
+<img src="../img/timeline/show-hidden-layer.gif" class="ms-docimage"  style="max-width:700px;" />
+
```
Comment (offtherailz) [inline docs/user-guide/timeline.md:65]: These options should be in JSDoc, no need to add a gif to explain this option, so you can remove it from the PR
```suggestion
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] Please document the configuration options in the JSDoc. 
- [ ] I see a strange behavior. when I toggle on the visibility of a time layer, it selects automcatically the new layer as the guide layer, while it should remain the same.
timeline

File: docs/user-guide/timeline.md (hunk 1 above)
Comment (dsuren1) [reply docs/user-guide/timeline.md:65]: Sure

**Review by offtherailz — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please test this in DEV and let us know if we can do the backport to the stable branch 2021.01.xx

**Comment by ElenaGallo** [comment]
@dsuren1 test done, can you backport this to stable branch? thanks

**Pull request merged**